src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -115,211 +92,136 @@ p5.RendererGL.prototype.endShape = function(
   isContour,
   shapeKind
 ) {
-  this._useImmediateModeShader();
-
-  if (this._doStroke && this.drawMode !== constants.TEXTURE) {
-    for (var i = 0; i < this.immediateMode.vertices.length - 1; i++) {
-      this.immediateMode.edges.push([i, i + 1]);
-    }
-    if (mode === constants.CLOSE) {
-      this.immediateMode.edges.push([
-        this.immediateMode.vertices.length - 1,
-        0
-      ]);
-    }
-
-    p5.Geometry.prototype._edgesToVertices.call(this.immediateMode);
-    this._drawStrokeImmediateMode();
-  }
-  if (this._doFill) {
-    this._drawFillImmediateMode(
-      mode,
-      isCurve,
-      isBezier,
-      isQuadratic,
-      isContour,
-      shapeKind
-    );
+  if (!this.isImmediateDrawing) {
+    throw new Error('endShape called without matching beginShape');
   }
-  //clear out our vertexPositions & colors arrays
-  //after rendering
-  this.immediateMode.vertices.length = 0;
-  this.immediateMode.vertexColors.length = 0;
-  this.immediateMode.uvCoords.length = 0;
-  this.isImmediateDrawing = false;
 
-  return this;
-};
+  var im = this.immediateMode;
+
+  if (this._doFill || this._doStroke) {
+    var gl = this.GL;
+    var drawMode;
+
+    if (this._doFill) {
+      // determine the draw mode to use
+      switch (this.shapeMode) {
+        case constants.POINTS:
+          drawMode = gl.POINTS;
+          break;
+
+        case constants.LINES:
+          drawMode = null;
+          break;
```

Comment:
Why is this a null draw mode? all of the others record the draw mode here.

Reply:
switch switch statement it selecting the draw mode to use for the fill. if the `shapeMode` is `LINES` then there's *no* (null) fill !